src/Riok.Mapperly/Descriptors/InlineExpressionMappingBuilderContext.cs

```diff
@@ -15,6 +16,7 @@ public class InlineExpressionMappingBuilderContext : MappingBuilderContext
 {
     private readonly MappingCollection _inlineExpressionMappings;
     private readonly MappingBuilderContext _parentContext;
+    private readonly ImmutableDictionary<(ISymbol, ISymbol), int> _parentTypes;
```

Comment:
Wouldn't this be much better suited inside the `MappingCollection`? Probably named as `mappingRecursionCount`. The mapping collection could then expose a method to get the recursion count of a mapping by its source and target type.

Reply:
IMO this makes more sense to be in `InlineExpressionMappingBuilderContext`.

`mappingRecursionCount` will either be unused by `MappingBuilderContext` or a new `MappingCollection` would have to be created for every context created to avoid passing state to other contexts.

Reply:
That makes sense, thank you for explaining.